addons/hr_holidays/models/hr_leave.py

```diff
@@ -650,17 +646,61 @@ def _inverse_supported_attachment_ids(self):
     def _check_date(self):
         if self.env.context.get('leave_skip_date_check', False):
             return
-        for holiday in self.filtered('employee_id'):
+
+        all_employees = self.employee_id | self.employee_ids
+        all_leaves = self.search([
+            ('date_from', '<', max(self.mapped('date_to'))),
+            ('date_to', '>', min(self.mapped('date_from'))),
+            ('employee_id', 'in', all_employees.ids),
+            ('id', 'not in', self.ids),
+            ('state', 'not in', ['cancel', 'refuse']),
+        ])
+        for holiday in self:
             domain = [
                 ('date_from', '<', holiday.date_to),
                 ('date_to', '>', holiday.date_from),
-                ('employee_id', '=', holiday.employee_id.id),
                 ('id', '!=', holiday.id),
                 ('state', 'not in', ['cancel', 'refuse']),
             ]
-            nholidays = self.search_count(domain)
-            if nholidays:
-                raise ValidationError(_('You can not set 2 time off that overlaps on the same day for the same employee.'))
+
+            employee_ids = (holiday.employee_id | holiday.employee_ids).ids
+            search_domain = domain + [('employee_id', 'in', employee_ids)]
+            conflicting_holidays = all_leaves.filtered_domain(search_domain)
+
+            if conflicting_holidays:
+                conflicting_holidays_list = []
+                conflicting_employees = set(employee_ids)
```

Comment:
I think you can do this, then get rid of line 685 with the same effect:

```suggestion
                conflicting_employees = set(employee_ids) - set(conflicting_holidays.employee_id.ids)
```

Reply:
This line can also be moved just before when it's needed, as it's not used in every cases -  this way it's only eval'ed when needed

Reply:
Done, thanks :-)